facilitator/src/bin/facilitator.rs

```diff
@@ -743,36 +848,218 @@ fn main() -> Result<(), anyhow::Error> {
+    let peer_name = &format!("{}-{}", locality_name, ingestor_name);
+    match bucket {
+        Some(bucket) => Ok((identity.map(String::from), String::from(bucket))),
+        None => {
+            let manifest_url =
+                manifest_url.expect("If bucket is not provided, manifest_url must be provided");
+            let manifest = SpecificManifest::from_https(manifest_url, peer_name).context(
+                format!("unable to read SpecificManifest from {}", manifest_url),
+            )?;
+
+            Ok((manifest.ingestion_identity(), manifest.ingestion_bucket()))
+        }
+    }
+}
+
+fn get_valid_batch_signing_key(
+    namespace: Option<&str>,
+    ingestor_manifest_url: Option<&str>,
+    matches: &ArgMatches,
+) -> Result<BatchSigningKey> {
+    match ingestor_manifest_url {
+        Some(own_manifest_url) => {
+            let namespace = namespace.unwrap();
+
+            let manifest =
+                IngestionServerManifest::from_https(own_manifest_url, None).context(format!(
+                    "unable to get ingestion server manifest from url: {}",
+                    own_manifest_url
+                ))?;
+
+            let kubernetes = Kubernetes::new(String::from(namespace));
+            let secrets = kubernetes.get_sorted_secrets("isrg-prio.org/type=batch-signing-key")?;
+
+            let batch_signing_keys = manifest.batch_signing_public_keys().unwrap();
+
+            let secret = secrets
+                .into_iter()
+                .find(|secret| batch_signing_keys.contains_key(&secret.name()));
+
+            match secret {
+                None => Err(anyhow!(
+            "unable to find a batch signing key from the manifest and kubernetes secret store"
+        )),
+                Some(secret) => {
+                    let secret_name = secret.name();
+                    let secret_data =
+                        base64::decode(secret.data.unwrap().remove("secret_key").unwrap().0)?;
+                    let key =
+                        EcdsaKeyPair::from_pkcs8(&ECDSA_P256_SHA256_ASN1_SIGNING, &secret_data)
+                            .map_err(|e| anyhow!("decoding secret key rejected: {}", e))?;
+
+                    Ok(BatchSigningKey {
+                        identifier: secret_name,
+                        key,
+                    })
+                }
+            }
+        }
+        // The caller is passing key in directly
+        None => batch_signing_key_from_arg(matches),
+    }
+}
 
 fn generate_sample(sub_matches: &ArgMatches) -> Result<(), anyhow::Error> {
-    let peer_output_path = StoragePath::from_str(sub_matches.value_of("peer-output").unwrap())?;
-    let peer_identity = sub_matches.value_of("peer-identity");
-    let packet_encryption_key = PrivateKey::from_base64(
-        sub_matches
-            .value_of("facilitator-ecies-private-key")
-            .unwrap(),
-    )
-    .unwrap();
+    let kube_namespace = sub_matches.value_of("kube-namespace");
```

Comment:
Does this mean I can't run `facilitator generate-ingestion-sample` on my laptop anymore? I'd still like to be able to generate sample data without running this in Kubernetes.

Reply:
You can, it doesn't require it.